src/index.ts

```diff
@@ -1,18 +1,43 @@
+import axios, {AxiosError, AxiosRequestConfig, AxiosResponse} from 'axios';
 import * as extend from 'extend';
-import * as _request from 'request';  // for types only
-import * as request from 'retry-request';
+import * as rax from 'retry-axios';
 
 const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
 
-export type Options = _request.CoreOptions&{property?: string, uri?: string};
+export type Options = AxiosRequestConfig&{property?: string, uri?: string};
 
 export type Callback =
-    (error: NodeJS.ErrnoException|null, response?: _request.RequestResponse,
+    (error: NodeJS.ErrnoException|null, response?: AxiosResponse<string>,
      metadataProp?: string) => void;
 
+// Accepts an options object passed from the user to the API.  In the
+// previous version of the API, it referred to a `Request` options object.
+// Now it refers to an Axios Request Config object.  This is here to help
+// ensure users don't pass invalid options when they upgrade from 0.4 to 0.5.
+// tslint:disable-next-line no-any
+function validate(options: any, callback: (err?: Error) => void) {
+  const vpairs = [
+    {invalid: 'uri', expected: 'url'}, {invalid: 'json', expected: 'data'},
+    {invalid: 'qs', expected: 'params'}
+  ];
+  for (const pair of vpairs) {
+    if (options[pair.invalid]) {
+      const e = `'${
+          pair.invalid}' is not a valid configuration option. Please use '${
+          pair.expected}' instead. This library is using Axios for requests. Please see https://github.com/axios/axios to learn more about the valid request options.`;
+      return callback(new Error(e));
+    }
+  }
+  callback();
+}
+
 export function _buildMetadataAccessor(type: string) {
   return function metadataAccessor(
-      options: string|Options|Callback, callback?: Callback) {
+      options?: string|Options|Callback, callback?: Callback) {
```

Comment:
Was making `options` optional intentional? Previously the compiler would have ensured that the callers pass in at least one parameter. To me it sounds like an error if the caller calls this function with no parameters.

Reply:
that was a mistake